cohort-engine/src/test/java/com/ibm/cohort/engine/parameter/ParameterTest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * (C) Copyright IBM Corp. 2021, 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.ibm.cohort.engine.parameter;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Test;
+import org.opencds.cqf.cql.engine.runtime.Interval;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class ParameterTest {
+	@Test
+	public void when_serialize_deserialize___properties_remain_the_same() throws Exception {
+		Parameter integer = new IntegerParameter(10);
+		Parameter decimal = new DecimalParameter("+100.99e10");
+		Parameter bool = new BooleanParameter(true);
+		Parameter str = new StringParameter("StringValueHere");
+		Parameter date = new DateParameter("2020-07-04");
+		Parameter datetime = new DatetimeParameter("2020-07-04T23:00:00-05:00");
+		Parameter time = new TimeParameter("@T23:00:00");
+
+		QuantityParameter quantity = new QuantityParameter("10", "mg/mL");
+		QuantityParameter denominator = new QuantityParameter("100", "mg/mL");
+		Parameter ratio = new RatioParameter(quantity, denominator);
+		Parameter end = new IntegerParameter(1000);
+		Parameter interval = new IntervalParameter(integer, true, end, false);
+		CodeParameter code = new CodeParameter("http://hl7.org/terminology/blob", "1234", "Blob", "1.0.0");
+		Parameter concept = new ConceptParameter("MyConcept", code);
+		
+		List<Parameter> parameters = Arrays.asList(integer, decimal, bool, str, date, datetime, time, quantity, denominator, ratio, interval, code, concept);
+		
+		ObjectMapper mapper = new ObjectMapper();
+		String serialized = mapper.writeValueAsString(parameters);
+		System.out.println(serialized);
+		assertFalse( serialized.contains("com.ibm") );
+		
+		List<Parameter> deserialized = mapper.readValue(serialized, new TypeReference<List<Parameter>>(){});
+		assertEquals( parameters.size(), deserialized.size() );
+		
+		for( int i=0; i<deserialized.size(); i++ ) {
+			Parameter expected = parameters.get(i);
+			Parameter actual = parameters.get(i);
```

Comment:
Is this supposed to be `deserialized.get(i)`?

Reply:
Great catch. Will fix